### PR TITLE
#2559. Add metadata tests

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: LanguageFeatures/Augmentation-libraries/metadata_A01_t01.dart [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  static String staticVariable = "Static variable";
+}
+
+extension type ET(int id) {
+  static String staticVariable = "Static variable";
+}
+
+main() {
+  Symbol emptyName = MirrorSystem.getSymbol('');
+  LibraryMirror libraryMirror = currentMirrorSystem().findLibrary(emptyName);
```
Comment (eernstg) [inline LanguageFeatures/Augmentation-libraries/metadata_A01_t01.dart:56]: What if there are multiple libraries whose name is empty? Is it really true that there aren't any? Perhaps it is safer to use `library some.name;` in this library and then find that.

Comment (eernstg) [reply LanguageFeatures/Augmentation-libraries/metadata_A01_t01.dart:56]: Same consideration seems to apply to several other tests.

File: LanguageFeatures/Augmentation-libraries/metadata_A01_t07.dart [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+/// @assertion All declarations can be augmented with metadata annotations
+/// and/or doc comments directly preceding an augmenting declaration.
+///
+/// In both cases, these should be appended to existing metadata or doc
+/// comments. For metadata annotations, these may trigger additional macro
+/// applications.
+///
+/// @description Check that augmenting metadata is appended to the introductory
+/// declaration. Test setters.
```
Comment (eernstg) [inline LanguageFeatures/Augmentation-libraries/metadata_A01_t07.dart:13]: Typo:
```suggestion
/// declaration. Test functions.
```

File: LanguageFeatures/Augmentation-libraries/metadata_A01_t08.dart [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+/// @assertion All declarations can be augmented with metadata annotations
+/// and/or doc comments directly preceding an augmenting declaration.
+///
+/// In both cases, these should be appended to existing metadata or doc
+/// comments. For metadata annotations, these may trigger additional macro
+/// applications.
+///
+/// @description Check that augmenting metadata is appended to the introductory
+/// declaration (not replaces it). Test setters.
```
Comment (eernstg) [inline LanguageFeatures/Augmentation-libraries/metadata_A01_t08.dart:13]: Typo:
```suggestion
/// declaration (not replaces it). Test functions.
```

**Review by eernstg — COMMENTED** [review]
Looks good! Just a couple of typos. I did suggest using a non-empty library name, too, because I'w worried that the empty library name would be ambiguous (if not now then later, when somebody wants to "clean up" a couple of SDK libraries with "unused" library names ;-).

**Comment by sgrekhov** [comment]
Updated. PTAL. Do we need a library name in an augmentating file?

**Comment by eernstg** [comment]
No. Every `augment library`  will soon be a part file, and they definitely do not allow for a library name. Also, 'dart:mirrors' should work on the contents of the library at the semantic level ("after merging"), not just the textual contents of the library file, which means that all references to the library should include the result of applying all augmentations, no matter which file they are located in.

**Review by eernstg — APPROVED** [review]
LGTM

**Pull request merged**